enhancements/machine-api/opt-out-machine-health-checking.md

```diff
@@ -0,0 +1,378 @@
+  - AWS attempts to launch instance, fails, puts in terminated state
+  - Machine controller marks Machine as Failed because instance is terminated
+- Conclusion:
+  - This scenario behaves in exactly the same way as the previous scenario
+
+### How does a Machine Health Check interact with Control-Plane Machine?
+
+At present, Machine Health Checks will not remediate Control-Plane machines.
+Since Control-Plane Machines are not owned by any controller, deleting Control-Plane machines would have to be recreated manually.
+
+Until automation is created (see [proposal](https://github.com/openshift/enhancements/pull/292))
+for management of Control-Plane machines, they should be excluded from Machine Health Checks as health checking them will be redundant.
+
+Once automation is created, a Machine Health Check could be used to remediate problems with the Control-Plane machines,
+though it must ensure that the Etcd cluster remains healthy and as such,
+should be managed by something to ensure the MaxUnhealthy parameter is set appropriately for the size of the Etcd cluster.
+
+### How does the MaxUnhealthy field work?
+
+Each MachineHealthCheck has a `maxUnhealthy` field.
+This field can be used to prevent the MachineHealthCheck from remediating problems with Machines if there are
+already too many unhealthy Machines within the cluster.
+
+For example, if the value of `maxUnhealthy` was `50%`, and in a 10 node cluster, 6 nodes were determined to be unhealthy,
+the `maxUnhealthy` value would prevent the MachineHealthCheck from making any remediation actions on the Machines.
+This mechanism prevents the MachineHealthCheck from potentially worsening a catastrophic or cascading failure with a cluster.
+
+An important note is that currently, the `maxUnhealthy` value is only respected for the single MachineHealthCheck that is being reconciled.
+This means that, if two MachineHealthChecks covered the same Machine or Machines, one of the MachineHealthChecks could have
+a less restrictive `maxUnhealthy` and allow a Machine to be remediated when the other MachineHealthCheck would have blocked remediation.
+
+This means that, if a MachineHealthCheck covered all nodes, the value will include the Control-Plane machines
+and they will be factored into the calculation to determine if MaxUnhealthy has been breached.
+Because of the special nature of Control-Plane Machines, including them in an MHC with other nodes that would potentially
+allow the MHC to remediate more than one Control-Plane machine at any one time (assuming three Control-Plane machines),
+could cause the cluster to lose quorum for Etcd and potentially even lose the data altogether.
+
+If Control-Plane Machines are to be covered by a MachineHealthCheck, the `maxUnhealthy` parameter must be appropriate to
+ensure that the MachineHealthCheck does not ever cause the Etcd cluster to lose quorum.
+For this reason, Control-Plane Machines should not be mixed with worker Machines within an MHC\*.
```

Comment:
is the "\*" leftover?

Reply:
Was meant to be like an asterisk addendum or note, will move to a brackets note to make more obvious, don't think it works in markdown as is